### DISCUSSION TITLE
Use Include Document feature on indexed fields?

### DISCUSSION THREAD

**Discussion by DanielToft** [ravendb/ravendb#14183] [Q&A] [answered]
Hallo

Is it not possible to use the Include document feature on an indexed field? 

I’m having the following index based on my “Effort" document. The effort document has a relation to a Workspace and the workspace document is related to my Asset document. The include feature do not seems to work on my asset document (WorkspaceImageId field).

Index:
``` c#
public class Effort_Test : AbstractIndexCreationTask<Effort, Effort_Test.Result>
{
    public class Result
    {
        public string Id { get; set; }
        public string WorkspaceId { get; set; }
        public string WorkspaceImageId { get; set; }
        // + other fields
    }

    public Effort_Test()
    {
        Map = efforts =>
            from effort in efforts
            let workspace = LoadDocument<Workspace>(effort.WorkspaceId)
            select new Result {
                Id = effort.Id,
                WorkspaceId = effort.WorkspaceId,
                WorkspaceImageId = workspace.Background.Image
            };
        
        Stores.Add(x => x.WorkspaceImageId, FieldStorage.Yes);
    }
}
```

Query:
``` c#
var results = await _session.Query<Effort_Test.Result, Effort_Test>()
                                .Include(x => x.WorkspaceId)
                                .Include(x => x.WorkspaceImageId)
                                // + other queries not related to the issue
                                .ProjectInto<Effort>() 
                                .ToListAsync();

foreach (var effort in results)
{
    // This will make make no requests to the server.
    var workspace = await _session.LoadAsync<Workspace>(effort.WorkspaceId); 
        
    // This will make an request to the server!
    var workspaceImage = await _session.LoadAsync<Asset>(workspace.Background.Image);
}
```

I would like to be able to query efforts and include all my relations, so I do not have to make multiple database request. 

I have thought about using projections, but then I will end up over fetching workspaces and assets. One workspace is related to many efforts (100-10.000) and one asset can be related to a few workspaces (1-10).

**Comment by ayende** [comment, marked as answer]
RavenDB will only do includes on values that are returned from the query.
When you call to `ProjectInto<Effort>()`, you are _not_ including `WorkspaceImageId` in the results.
As such, it won't be included. Make sure that you have the field in the output. Can inherit from Effect and add that field, for example.

**Comment by DanielToft** [comment]
Thanks @ayende! That works perfectly.
